### PR TITLE
[Feature] Media channel support

### PR DESCRIPTION
**Pull request by Misha-133**
[docs](https://github.com/discord/discord-api-docs/pull/6232)

This PR adds support for the new media channels. 

## Warning
This feature is not released & fully documented yet, so use this at your own risk as things might change at any moment.

### Changes
- add `IMediaChannel`, rest & socket entities.
- add new channel flag
- add `IsThumbnail` property to `FileAttachment` - one can be set for an image used as a thumbnail in a media channel post
- update everything to support new `ChannelType`
- **misc** Add some missing forum channel related methods to `IGuild`

#### This PR is based on PR #2720

### CODE REVIEW

File: src/Discord.Net.Rest/Entities/Guilds/RestGuild.cs [hunk 1]
```diff
@@ -726,11 +784,24 @@ public Task<RestCategoryChannel> CreateCategoryChannelAsync(string name, Action<
         /// <param name="options">The options to be used when sending the request.</param>
         /// <exception cref="ArgumentNullException"><paramref name="name" /> is <see langword="null"/>.</exception>
         /// <returns>
-        ///     The created category channel.
+        ///     The created forum channel.
```
Comment (Panthr75) [inline src/Discord.Net.Rest/Entities/Guilds/RestGuild.cs:787]: Rename `category` on line 780 to `forum`, like here

File: src/Discord.Net.WebSocket/Entities/Guilds/SocketGuild.cs [hunk 2]
```diff
@@ -337,6 +337,15 @@ public IReadOnlyCollection<SocketThreadChannel> ThreadChannels
         public IReadOnlyCollection<SocketForumChannel> ForumChannels
             => Channels.OfType<SocketForumChannel>().ToImmutableArray();
```
Comment (Panthr75) [inline src/Discord.Net.WebSocket/Entities/Guilds/SocketGuild.cs:338]: Might want to add a check to not include media channels here

**Comment by Panthr75** [comment]
It would be nice to have a `MediaChannelProperties`, so that you aren't passing `ForumChannelProperties` to a `Create Media Channel` method -- which seems wrong.

File: src/Discord.Net.Rest/Entities/Guilds/RestGuild.cs (hunk 1 above)
Comment (Misha-133) [reply src/Discord.Net.Rest/Entities/Guilds/RestGuild.cs:787]: ah yup, nice catch

File: src/Discord.Net.WebSocket/Entities/Guilds/SocketGuild.cs (hunk 2 above)
Comment (Misha-133) [reply src/Discord.Net.WebSocket/Entities/Guilds/SocketGuild.cs:338]: This way it matches the behavior of similar methods.
ex: `SocketGuild.TextChannels`, `SocketGuild.VoiceChannels`

**Comment by Misha-133** [comment]
> It would be nice to have a `MediaChannelProperties`, so that you aren't passing `ForumChannelProperties` to a `Create Media Channel` method -- which seems wrong.

I don't see a reason to add more properties, since the media channel does not have any extra features. Also, this matches the `IStageChannel` reusing same `VoiceChannelProperties`

**Review by MrCakeSlayer — APPROVED** [review]

**Review by quinchs — APPROVED** [review]

**Pull request merged**